region/maridia/inner-green/Shaktool Room.json

```diff
@@ -646,6 +676,87 @@
         "Grapple teleporting crosses the room, so it will set the flag 'f_ShaktoolDoneDigging', assuming that the game is modified to treat the room symmetrically with how the flag gets set."
       ]
     },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Entry",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "clearsObstacles": ["R-Mode"]
+    },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Entry - Shaktool Done Digging",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [
+        "f_ShaktoolDoneDigging"
+      ],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "clearsObstacles": ["A", "B", "R-Mode"],
+      "devNote": "Entering the room with the flag 'f_ShaktoolDoneDigging' set means that the sand blocks and Power Bomb blocks will not spawn."
+    },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit)",
+      "requires": [
+        {"obstaclesCleared": ["R-Mode", "B"]},
+        {"or": [
+          "h_CrystalFlashForReserveEnergy",
+          {"and": [
+            "h_RModeCanRefillReserves",
+            {"or": [
+              {"and": [
+                {"obstaclesCleared": ["A"]},
+                {"or": [
+                  "canWaterShineCharge",
+                  "Gravity"
+                ]},
+                "h_shinechargeMaxRunway",
+                {"shinespark": {"frames": 4, "excessFrames": 0}},
+                {"or": [
+                  {"and": [
+                    {"resourceMissingAtMost": [{"type": "Missile", "count": 0}]},
+                    {"partialRefill": {"type": "ReserveEnergy", "limit": 60}}
+                  ]},
+                  {"partialRefill": {"type": "ReserveEnergy", "limit": 20}}
+                ]}
+              ]},
+              {"and": [
+                "canBePatient",
+                {"enemyKill": {"enemies": [["Yard"]]}},
```

Comment:
As is, this doesn't include killing Yards with Speed.  And that would need a explanation as well.

Reply:
The number of times I've accidentally run through and killed a snail in Aqueduct may have made me think it was more obvious than it actually is. I'll check this some more.

Using Bombs + Charge Beam might also help for a safer bomb kill.